python/cudf/cudf/core/series.py

```diff
@@ -971,15 +971,14 @@ def __repr__(self):
             preprocess = self.copy()
 
         preprocess.index = preprocess.index._clean_nulls_from_index()
-        if isinstance(preprocess._column, cudf.core.column.TimeDeltaColumn):
-            preprocess._column = preprocess._column._repr_str_col()
-            output = preprocess.to_pandas().__repr__()
-        elif (
+        if (
             preprocess.nullable
             and not isinstance(
                 preprocess._column, cudf.core.column.CategoricalColumn
             )
             and not is_list_dtype(preprocess.dtype)
+        ) or isinstance(
+            preprocess._column, cudf.core.column.timedelta.TimeDeltaColumn
```

Comment:
```suggestion
```

LGTM overall, I think without this condition we should be able to run into this block if we have nulls, else we will take the else block.

Reply:
Yeah but I think even when there are no nulls we still need to cast to string to get the right result. Otherwise it hits `to_pandas()` first and it can change the formatting of the values. 

```
0    0 days 00:00:00.001000000
1    0 days 00:00:00.000200000
2    0 days 00:00:00.003000000
dtype: timedelta64[ns]
(Pdb) cp.asnumpy(ser._column.values)
array([1000000,  200000, 3000000], dtype='timedelta64[ns]')
(Pdb) pd.Series(cp.asnumpy(ser._column.values), dtype=ser.dtype)
0   0 days 00:00:00.001000
1   0 days 00:00:00.000200
2   0 days 00:00:00.003000
dtype: timedelta64[ns]
```

Reply:
Ah yes, you are correct. Totally forgot this difference.